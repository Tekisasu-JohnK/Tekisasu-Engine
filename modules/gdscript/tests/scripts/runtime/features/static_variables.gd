@static_unload

static var perm := 0

static var prop := "Hello!":
	get: return prop + " suffix"
	set(value): prop = "prefix " + str(value)

static func get_data():
	return "data"

static var data = get_data()

class Inner:
	static var prop := "inner"
	static func _static_init() -> void:
		prints("Inner._static_init", prop)

	class InnerInner:
		static var prop := "inner inner"
		static func _static_init() -> void:
			prints("InnerInner._static_init", prop)

func test():
	prints("data:", data)

	prints("perm:", perm)
	prints("prop:", prop)

	perm = 1
	prop = "World!"

	prints("perm:", perm)
	prints("prop:", prop)

	print("other.perm:", StaticVariablesOther.perm)
	print("other.prop:", StaticVariablesOther.prop)

	StaticVariablesOther.perm = 2
	StaticVariablesOther.prop = "foo"

	print("other.perm:", StaticVariablesOther.perm)
	print("other.prop:", StaticVariablesOther.prop)

	@warning_ignore("unsafe_method_access")
	var path = get_script().get_path().get_base_dir()
	var other = load(path  + "/static_variables_load.gd")

	print("load.perm:", other.perm)
	print("load.prop:", other.prop)

	other.perm = 3
	other.prop = "bar"

	print("load.perm:", other.perm)
	print("load.prop:", other.prop)
