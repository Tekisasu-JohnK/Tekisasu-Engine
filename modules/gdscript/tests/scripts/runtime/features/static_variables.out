GDTEST_OK
Inner._static_init inner
InnerInner._static_init inner inner
data: data
perm: 0
prop: prefix Hello! suffix
perm: 1
prop: prefix World! suffix
other.perm:0
other.prop:prefix Hello! suffix
other.perm:2
other.prop:prefix foo suffix
load.perm:0
load.prop:prefix Hello! suffix
load.perm:3
load.prop:prefix bar suffix
